src/uct/ib/base/ib_iface.c

```diff
@@ -177,7 +177,22 @@ int uct_ib_iface_is_reachable(const uct_iface_h tl_iface, const uct_device_addr_
     uint16_t lid;
 
     uct_ib_address_unpack(ib_addr, &lid, &is_global, &gid);
-    return (gid.global.subnet_prefix == iface->gid.global.subnet_prefix);
+
+    if ((ib_addr->flags & UCT_IB_ADDRESS_FLAG_GID_RAW) == 0) {
```

Comment:
1. !(ib_addr->flags & UCT_IB_ADDRESS_FLAG_GID_RAW)
2. need to check that the local link layer type is matching the remote one